scripts/src/public/services/orders.ts

```diff
@@ -520,3 +520,40 @@ function checkIfTimedOut(order: db.Order): Promise<void> {
 function getLockResource(type: "create" | "get", ...ids: string[]): string {
 	return `locks:orders:${ type }:${ ids.join(":") }`;
 }
+
+export async function createCrossAppOrder(recipientWalletAddress: string, appId: string, title: string, description: string, amount: number, sender: User, senderDeviceId: string): Promise<OpenOrder> {
+
+	logger().info("creating a cross app order")
+
+	const senderWallet = (await sender.getWallets(senderDeviceId)).lastUsed();
+	if (!senderWallet) {
+		throw UserHasNoWallet(sender.id, senderDeviceId);
+	}
+
+	if (await WalletApplication.getBlockchainVersion(senderWallet.address) !==
+		await WalletApplication.getBlockchainVersion(recipientWalletAddress)) {
```

Comment:
According to Berry this feature will only ever be released to KIN3. You don't have to check this here. Furthermore, if you check this you might not get what you expect when one of the apps is not an ecosystem app (and the wallet isn't in the DB).

Reply:
removed